### PR TITLE
⚠️ do not merge/arrowsquid ahp

### PR DESCRIPTION
**Pull request by vikiival**
- **:wrench: arrow stick AHP**

**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [ ] Closes #<issue_number>
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?

- [ ] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=<My_Polkadot_Address_check_https://github.com/kodadot/nft-gallery/blob/main/REWARDS.md#creating-your-dot-address>)

## Screenshot 📸

- [ ] My fix has changed UI

### CODE REVIEW

**Comment by yangwao** [comment]
@vikiival, what are you waiting for here?

**Comment by vikiival** [comment]
- https://github.com/kodadot/stick/issues/220#issuecomment-1989040942

Made another release

**Comment by vikiival** [comment]
If it is OK let's close it then

**Comment by vikiival** [comment]
Ah I forgot to enable ws, thanks for the test

**Comment by vikiival** [comment]
issue was in the trailing slash [e73dc2a](https://github.com/kodadot/nft-gallery/pull/9577/commits/e73dc2afec3ff175248e5a70fbcb5f40768abce8)

**Comment by yangwao** [comment]
Needs to be tested and then can be promoted to production?

**Comment by vikiival** [comment]
> Needs to be tested and then can be promoted to production? 

Exactly, once we close this PR i will promote arrowsquid to prod

**Comment by prury** [comment]
@vikiival i'm getting different query results: 

image

**Comment by vikiival** [comment]
> @vikiival i'm getting different query results:

I have missed the PR from @daiagi 

- https://github.com/kodadot/stick/pull/251

Should be fixed in ~30 min

**Comment by vikiival** [comment]
Probably fixed ? 

kudos to @daiagi 
Screenshot 2024-03-22 at 13 52 17

**Comment by yangwao** [comment]
putting it live?

**Comment by prury** [comment]
> Probably fixed ?
> 
> kudos to @daiagi Screenshot 2024-03-22 at 13 52 17

yup, this is fixed

only small difference in query results now (21) items

**Comment by vikiival** [comment]
> only small difference in query results now (21) items on explore -> items

I am not sure what you mean. Can you provide screenshots please?

**Comment by vikiival** [comment]
> Same as this, but with smaller difference

I suppect there is a huge hidden bug in the token entity (cc @daiagi )

**Comment by daiagi** [comment]
Should be some sql issue in the total token entity resolver, nothing major (in terms of effort to fix) 
Can assume that previously number is the correct one as I checked it well when I first wrote it. And the previous query was simple

**Comment by daiagi** [comment]
Something completely different is going on here and it's not so easy to explain over gh post

1. it isn't related to tokenEntity at all, and token entity is reporting correct numbers (compare PR and canary with "the break" search query)

2. it's about metadata mismatch between canary and pr env. no idea why. maybe because the indexer is stuck?


let's walk through it, we will focus on Motifs collection

1. in explore page, search for "motifs"
2. notice that `count@pr = count@canary +4`
3. scroll to bottom and find a stacked nft with countof 8@canary and count of 4@PR. this account for all the mismatch
4. go to collection page (of motifs), collection page does not use TokenEntity at all, notice that total count is same in both envs
5. manually count the amount of stacked nfts, you will find that the count on the "stacked" card was individually correct in each env
6. think to yourself wtf?
7. find out that nft `108-88`, shows the default drop picture on canary (stacked) and **NOT** the default picture on PR env (not stacked)


image



**tl;dr**


some nfts have default image on canary and not default on PR env, which leads to a different count in explore page due to "stacked" cards @vikiival  ant thoughts? my only idea is due to stuck indexer

also, canary env does not impl yet the "tokenEntity" id change, that fixed "tokenEntity" bunching together nfts that should not be bunched together, that also leads to count mismatch

**Comment by vikiival** [comment]
> also, canary env does not impl yet the "tokenEntity" id change, that fixed "tokenEntity" bunching together nfts that should not be bunched together, that also leads to count mismatc

Except this line I was able to understand the issue you have described above ^-^, thanks for the explanation.

probably this would be a correct thing to do 
- https://github.com/kodadot/stick/issues/255

**Comment by vikiival** [comment]
Side effects are my favs 

Screenshot 2024-03-29 at 14 07 01
